pkg/types/metadata.go

```diff
@@ -124,6 +129,15 @@ func (m Metadata) Root() Metadata {
 	return *meta
 }
 
+func (m Metadata) WithInternal(internal interface{}) Metadata {
+	m.internal = internal
```

Comment:
Can we get rid of these `Internal` functions now? I can't remember what they were for but I don't think we're using them now?